packages/ckeditor5-paste-from-office/src/filters/table.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module paste-from-office/filters/parse
+ */
+import type { UpcastWriter, ViewDocumentFragment } from 'ckeditor5/src/engine';
+
+/**
+ * Set alignment for table pasted from MS Word.
+ *
+ * @param documentFragment The view structure to be transformed.
+ */
+export function tableAlignmentFilter( documentFragment: ViewDocumentFragment, writer: UpcastWriter ): void {
+	for ( const item of documentFragment.getChildren() ) {
+		if ( !item.is( 'element' ) ) {
+			continue;
+		}
+
+		// If table is not wrapped in div[align] it should be alligned left.
+		// More details:(https://github.com/ckeditor/ckeditor5/issues/8752#issuecomment-1623507171).
+		if ( item.is( 'element', 'table' ) ) {
+			writer.setAttribute( 'align', 'left', item );
+			continue;
+		}
+
+		const align = item.getAttribute( 'align' );
+		const child = item.getChild( 0 );
+
+		// Element should have alignment attribute and should have child and should be div.
+		if ( !align || !child || item.name !== 'div' ) {
+			continue;
+		}
+
+		// If table is wrapped in div[align] table should be alligned with defined alignment value.
+		// More details:(https://github.com/ckeditor/ckeditor5/issues/8752#issuecomment-1629065676)
+		if ( child.is( 'element', 'table' ) ) {
+			writer.setAttribute( 'align', align === 'center' ? 'none' : align, child );
```

Comment:
I'm unsure about setting the `align` attribute on view `table` elements. As per [https://github.com/ckeditor/ckeditor5/issues/8752#issuecomment-1629065676:](https://github.com/ckeditor/ckeditor5/issues/8752#issuecomment-1629065676:)

> *   that has a parent element `div[align=left]` or `div[align=right]` then the model's `table` element should get `tableAlignment` set to `left` or `right`.

If you apply it as you did, I'm not sure it will work with the table properties as it uses the mentioned `tableAlignment` property in model and styling (`float: right/left`) in view. Pasted table should have the correct alignment selected.

![](https://github.com/ckeditor/ckeditor5/assets/10883184/a68c25a7-41e1-44d4-993d-4e3c471afa54)

Reply:
With `tableAlignment` it doesn't work, I guess because in upcast you could just apply DOM-like attributes.